### PR TITLE
Adding AES implementation in python

### PR DESCRIPTION
**Pull request by Salonii02**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
YES

### Description

Adding implementation of AES algorithm in Python

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

AES implementation in Python #2098

### CODE REVIEW

**Comment by Salonii02** [comment]
Hello @anushkrishnav,
I have added AES implementation in python first using pycrypto and later using pycryptodome since security checks were not successful. Library mentioned in your analysis check is pyca/cryptography for which I will have to implement in a new manner.
can you guide me here or using pycrypto or pycryptodome library is fine.

**Comment by Salonii02** [comment]
@atarax665 @anushkrishnav kindly check.

File: Python/cryptography/AES.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+# Enter the block-size in bytes: 16
+# Enter your msg to test algorithm: I love cats
+# Encrypted_text is:b'OomGSg7EzTt6EYKdrtDMQW31R7tNw5Y4ns9sbiYsIHM='
+# Decrypted_text is: I love cats
+
+# Enter your secret key: qazxswedcplmnko
+# Enter the block-size in bytes: 16
+# Enter your msg to test algorithm: I love AES-algo
+# Encrypted_text is:b'11f9GkCARXrbcMibADUR7wElYU5lLdIexwhEJaXu2Xc='
+# Decrypted_text is: I love AES-algo
```
Comment (HarshCasper) [inline Python/cryptography/AES.py]: Put the same in a multi-line string

File: Python/cryptography/AES.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+# Testing algorithm
+# asking for the message to test
+input_text = input("Enter your msg to test algorithm: ")
+cipher_text = encryption(input_text)
+print("Encrypted_text is: ")
+print(cipher_text)
+decrypted_text = decryption(cipher_text)
+print("Decrypted_text is: ")
+print(decrypted_text)
```
Comment (HarshCasper) [inline Python/cryptography/AES.py]: Put this in a `if __name__ == '__main__'` block of Code

File: Python/cryptography/AES.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    # converting message from characters to bits
+    message = base64.b64decode(message)
+    # extracting our intializationvector which will be first block_size bits of message
+    initialize_vector = message[:block_size]
+    # creating new cipher with our hash_key,in mode CBC and extracted intialized vector
+    cipher = AES.new(hash_key, AES.MODE_CBC, initialize_vector)
+    # envoking decrypt function from new cipher and decoding it into text
+    plain_text = cipher.decrypt(message[block_size:]).decode("utf-8")
+    # finally unpadding plain_text and returning
+    return unpadding(plain_text)
```
Comment (HarshCasper) [inline Python/cryptography/AES.py]: Use proper Docstrings instead of putting comments for every line of code

File: Python/cryptography/AES.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    # generating a new random initialization_vector with the size of an AES block
+    initialize_vector = Random.new()
+    initialize_vector = initialize_vector.read(block_size)
+    # creating AES cipher with our hash-key,in mode CBC and initialized vector
+    cipher = AES.new(hash_key, AES.MODE_CBC, initialize_vector)
+    # envoking encrypt function from generated cipher
+    encrypted_text = cipher.encrypt(message.encode())
+    # placing encrypted_text after initialization_vector and
+    # converting back from bits to readable characters.
+    return base64.b64encode(initialize_vector+encrypted_text)
```
Comment (HarshCasper) [inline Python/cryptography/AES.py]: Use proper Docstrings instead of putting comments for every line of code

File: Python/cryptography/AES.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+def padding(msg, blocks_size):
+    padding_bytes_len = blocks_size - len(msg) % blocks_size
+# The chr() method returns a character from an integer (unicode)
+# we are generating padding char using padding_bytes_len
+# we can also use any padding char as explained in above example such as -,= etc
+    padded_char = chr(padding_bytes_len)
+    padding_string = padding_bytes_len*padded_char
+    padded_res = msg+padding_string
+    return padded_res
```
Comment (HarshCasper) [inline Python/cryptography/AES.py]: Use proper Docstrings instead of putting comments for every line of code

File: Python/cryptography/AES.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+def unpadding(msg):
+    last_char = msg[len(msg) - 1:]
+# The ord() function returns the number representing
+# the unicode code of a character.
+# as we added unicode code of character to make
+# it padded we are removing same.
+    unpadded_bytes_len = ord(last_char)
+    unpadded_res = msg[:-unpadded_bytes_len]
+    return unpadded_res
```
Comment (HarshCasper) [inline Python/cryptography/AES.py]: Use proper Docstrings instead of putting comments for every line of code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Python/cryptography/AES.py (hunk 6 above)
Comment (Salonii02) [reply Python/cryptography/AES.py]: I have pushed requested changes. kindly check @HarshCasper

**Comment by Salonii02** [comment]
I have pushed requested changes. kindly check @HarshCasper

**Comment by Salonii02** [comment]
Hello @HarshCasper ,
please review the requested changes.

**Review by HarshCasper — APPROVED** [review]

**Comment by Salonii02** [comment]
Hello @HarshCasper, can you please add the required labels for DWOC?
Thank you.

**Comment by Salonii02** [comment]
Hello @HarshCasper, I request you to add levels tags,
Thank you

**Pull request merged**